domain_config/src/Tests/DomainConfigOverriderTest.php

```diff
@@ -54,6 +54,34 @@ public function testDomainConfigOverrider() {
   }
 
   /**
+   * Tests that domain-specific variable overrides in settings.php works.
+   */
+  public function testDomainConfigOverriderFromSettings() {
+    // Set up overrides.
+    $settings = [];
+    $settings['config']['domain.config.one_example_com.en.system.site']['name'] = (object) [
+      'value' => 'First',
+      'required' => TRUE,
+    ];
+    $settings['config']['domain.config.four_example_com.system.site']['name'] = (object) [
+      'value' => 'Four overridden in settings',
+      'required' => TRUE,
+    ];
+    $this->writeSettings($settings);
```

Comment:
As per function [docbloc](http://cgit.drupalcode.org/drupal/tree/core/modules/simpletest/src/WebTestBase.php?h=8.1.x#n920)

```php
  /**
   * Rewrites the settings.php file of the test site.
   *
   * @param array $settings
   *   An array of settings to write out, in the format expected by
   *   drupal_rewrite_settings().
   *
   * @see drupal_rewrite_settings()
   */
  protected function writeSettings(array $settings) {
    include_once DRUPAL_ROOT . '/core/includes/install.inc';
    $filename = $this->siteDirectory . '/settings.php';
    // system_requirements() removes write permissions from settings.php
    // whenever it is invoked.
    // Not using File API; a potential error must trigger a PHP warning.
    chmod($filename, 0666);
    drupal_rewrite_settings($settings, $filename);
  }
```

and `drupal_write_settings()` does the actual writing to file, invalidates the opcode cache and loads the new settings in a `Settings` object.

So I believe we are testing the correct thing.